specs/request-handler-plugins.md

```diff
@@ -0,0 +1,116 @@
+# Support a "plug-in" model for request handlers
+
+To allow developers to provide unique handling behaviors for different APIs the Microsoft Graph will have a plug-in model to dynamically load request handlers. This will provide separation of concerns between the software components intercepting network request and those providing responses.
+To enable developers building handlers an abstractions library will be distributed on NuGet.
+
+## History
+
+| Version | Date | Comments | Author |
+| ------- | -------- | ----- | --- |
+| 1.0 | 2022-12-19 | Initial specifications | @gavinbarron |
+
+## Configuration file
+
+The existing `appsettings.json` file will be used to provide configuration. Both for determining which request handlers are loaded into the proxy and providing configuration for each of the loaded request handlers. The set of request handlers to be loaded will be specified as an array of objects at the root of the configuration file in the `handlers` property. Each handler must supply a `configSection`, `handerPath`, `name`, and `urlsToWatch` in the handlers array.
```

Comment:
Shall we name the collection `plugins` rather than `handlers` just so that we cater to a broader set of scenarios, beyond what we're envisioning now? Also, aren't handlers referring just to the code that handles an intercepted request, whereas we also wanted to support behaviors, like rate limiting?

Reply:
I like the name shift. 
Rate limiting is still going to need to intercept the requests to apply the metering, and potentially apply a response if the meter is exceeded.